tests/test_umls_api.py

```diff
@@ -0,0 +1,172 @@
+    with raises:
+        responses.add(responses.GET, AUTH_URL, body=res_text, status=res_status)
+        api = umls.UmlsApi(api_key, validator_key)
+        assert api.api_key == api_key
+        if validator_key is not None:
+            assert api.validator_key == validator_key
+        else:
+            assert api.validator_key == api_key
+
+
+@mock.patch.dict(
+    os.environ,
+    clear=True,
+)
+@responses.activate
+def test_auth_reads_env_var():
+    responses.add(responses.GET, AUTH_URL, body="true", status=200)
+    os.environ["UMLS_API_KEY"] = "valid_key"
+    api = umls.UmlsApi()
+    assert api.api_key == "valid_key"
+
+
+def get_valueset_data(file_name):
+    file_path = pathlib.Path(f"./tests/test_data/apis/umls/{file_name}")
+    with open(file_path) as f:
+        return f.read()
+
+
+@mock.patch.dict(
+    os.environ,
+    clear=True,
+)
+@pytest.mark.parametrize(
+    "url,oid,expected_oids,raises",
+    [
+        (VALUESET_URL, SINGLE_VALUESET_OID, [SINGLE_VALUESET_OID], does_not_raise()),
+        (None, SINGLE_VALUESET_OID, [SINGLE_VALUESET_OID], does_not_raise()),
+        (
+            VALUESET_URL + "/" + SINGLE_VALUESET_OID,
+            None,
+            [SINGLE_VALUESET_OID],
+            does_not_raise(),
+        ),
+        (
+            VALUESET_URL,
+            INCLUDE_VALUESET_OID,
+            [INCLUDE_VALUESET_OID, SINGLE_VALUESET_OID],
+            does_not_raise(),
+        ),
+        (
+            None,
+            INCLUDE_VALUESET_OID,
+            [INCLUDE_VALUESET_OID, SINGLE_VALUESET_OID],
+            does_not_raise(),
+        ),
+        (
+            VALUESET_URL + "/" + INCLUDE_VALUESET_OID,
+            None,
+            [INCLUDE_VALUESET_OID, SINGLE_VALUESET_OID],
+            does_not_raise(),
+        ),
+        (None, None, [], pytest.raises(errors.ApiError)),
+    ],
+)
+@responses.activate
+def test_get_valueset(url, oid, expected_oids, raises):
+    with raises:
+        responses.add(responses.GET, AUTH_URL, body="true", status=200)
+        responses.add(responses.GET, VALUESET_URL, status=404)
+        responses.add(
+            responses.GET,
+            VALUESET_URL + "/" + SINGLE_VALUESET_OID,
+            body=get_valueset_data("single_valueset.json"),
+            status=200,
+        )
+        responses.add(
+            responses.GET,
+            VALUESET_URL + "/" + INCLUDE_VALUESET_OID,
+            body=get_valueset_data("include_valueset.json"),
+            status=200,
+        )
+        api = umls.UmlsApi(api_key="123")
+        data = api.get_vsac_valuesets(url=url, oid=oid)
+        assert len(data) == len(expected_oids)
+        for i in range(0, len(expected_oids)):
+            assert data[i]["id"] == expected_oids[i]
+
+
+@mock.patch.dict(
+    os.environ,
+    clear=True,
+)
+@responses.activate
+def test_download_umls(tmp_path):
+    with open("./tests/test_data/apis/umls/umls.zip", "rb") as download_zip:
```

Comment:
Can you add a comment here or somewhere about what the origin of this zip file is / how to re-generate it if needed or whatever.

Reply:
yeah, can do.